sirepo/package_data/template/srw/parameters.py.jinja

```diff
@@ -376,94 +366,24 @@ def _read_srw_file(filename):
     return data
 
 
-def _rsopt_run(filename):
-    from pykern import pkio
-    import h5py
-    import multiprocessing
-    import time
-
-    # create temporary directories
-    for d in _TMP_DIRS:
-        pkio.mkdir_parent(d)
-
-    start_time = time.time()
-    data = numpy.load(filename)
-    tasks = data['x']
-    shape = data.dtype['x'].shape
-    # if a single parameter varies, rsopt does not put it into an array, so we must
-    if not shape:
-        tasks = tasks.reshape(len(tasks), -1)
-    n_runs = len(tasks)
-
-    n_processes = min(n_runs, multiprocessing.cpu_count() - 1)
-    pkdlog(f'Number of processes available: {n_processes}')
-
-    ####################### splits up the number of tasks to be done evenly across the number of processes available and assigns a task cut to each process
-    processes = []
-    split = n_runs // n_processes
-    idx0 = 0
-    for j in range(n_processes):
-        idx1 = n_runs if j == n_processes - 1 else idx0 + split
-        p = multiprocessing.Process(target=_rsopt_run_set, args=(tasks, idx0, idx1, j))
-        processes.append(p)
-        p.start()
-        idx0 += split
-
-    pkdlog('Processes started')
-
-    ####################### must join all processes to end multiprocess thread
-    for p in processes:
-        pkdlog(f'joining {p}...')
-        p.join()
-        pkdlog(f'joined {p}')
-
-    end_time = time.time()
-    pkdlog(f'Time to run {n_runs} simulations with {n_processes} processes: {numpy.round((end_time - start_time) / 60, 4)}m')
-
-    ###### reads in the temporary storage folder info to create one input and one output file for ML
-    beam_arrays = []
-    task_arrays = []
-    for i in range(n_runs):
-        beam_arrays.append(numpy.load(f'{_BEAM_TMP_DIR}/beam_{i}.npy'))
-        task_arrays.append(numpy.load(f'{_PARAM_TMP_DIR}/values_{i}.npy'))
-
-    beams_all = numpy.concatenate(beam_arrays, axis=0)
-    params_all = numpy.concatenate(task_arrays, axis=0)
-
-    ###### consolidated input and output file used for ML
-    with h5py.File(f'{_DATASET_DIR}/results.h5', 'w') as f:
-        f.create_dataset('params', data=[{{ rsOptFuctionSignature(true) }}])
-        f.create_dataset('paramVals', data=params_all.tolist())
-        f.create_dataset('beamIntensities', data=beams_all.tolist())
-
-def _rsopt_run_set(param_arr, start, end, proc_num):
-    pkdlog(f'Process {proc_num} to complete {end - start} tasks')
-    task_num = start
-    param_set = param_arr[start:end]
-    for vals in param_set:
-        _rsopt_run_single(vals, proc_num, task_num)
-        task_num += 1
-
-
-def _rsopt_run_single(param_vals, proc_num, task_num):
-    vp = _rsopt_set_params(*param_vals)
+def rsopt_run({{ rsOptFuctionSignature() }}):
+    vp = _rsopt_set_params({{ rsOptFuctionSignature() }})
```

Comment:
Unless I am mistaken, it looks like `vp` is declared and only used once. If so maybe the variable is not necessary and you could call
```python
f = _get_beamline_param(
    _rsopt_set_params({{ rsOptFuctionSignature() }}),
    'ws_fni',
    '{{ ws_fni_desc }}'
)

?

Reply:
It's a little unclear, but `{{srwMain}}`, references `vp` once jinja expands it.